pkg/registry/entry_test.go

```diff
@@ -0,0 +1,341 @@
+		},
+		{
+			e:   ParseEntry("192.168.1.1"),
+			s:   "192.168.1.1:123",
+			res: false,
+		},
+		{
+			e:   ParseEntry("http://192.168.1.1"),
+			s:   "http://192.168.1.1",
+			res: true,
+		},
+		{
+			e:   ParseEntry("http://192.168.1.1"),
+			s:   "192.168.1.1/foo/bar",
+			res: false,
+		},
+		{
+			e:   ParseEntry("http://192.168.1.1"),
+			s:   "https://192.168.1.1",
+			res: false,
+		},
+		{
+			e:   ParseEntry("https://192.168.1.1"),
+			s:   "http://192.168.1.1",
+			res: false,
+		},
+	}
+
+	for _, te := range tests {
+		assert.Equal(t, te.res, te.e.Match(te.s), "test: %s match %s", te.e, te.s)
+	}
+}
+
+func TestEntryEqual(t *testing.T) {
+	var tests = []struct {
+		e, other Entry
+		res      bool
+	}{
+		{
+			e:     ParseEntry("192.168.0.1"),
+			other: ParseEntry("192.168.0.1"),
+			res:   true,
+		},
+		{
+			e:     ParseEntry("192.168.0.1"),
+			other: ParseEntry("192.168.0.2"),
+			res:   false,
+		},
+		{
+			e:     ParseEntry("192.168.0.1"),
+			other: ParseEntry("192.168.1.0/24"),
+			res:   false,
+		},
+		{
+			e:     ParseEntry("192.168.0.1"),
+			other: ParseEntry("*.google.com"),
+			res:   false,
+		},
+		{
+			e:     ParseEntry("192.168.0.1/24"),
+			other: ParseEntry("192.168.0.1/24"),
+			res:   true,
+		},
+		{
+			e:     ParseEntry("192.168.0.1/24"),
+			other: ParseEntry("192.168.0.1/16"),
+			res:   false,
+		},
+		{
+			e:     ParseEntry("192.168.0.1/24"),
+			other: ParseEntry("192.168.0.1"),
+			res:   false,
+		},
+		{
+			e:     ParseEntry("192.168.0.1/24"),
+			other: ParseEntry("*.google.com"),
+			res:   false,
+		},
+		{
+			e:     ParseEntry("*.google.com"),
+			other: ParseEntry("*.google.com"),
+			res:   true,
+		},
+		{
+			e:     ParseEntry("*.google.com"),
+			other: ParseEntry("mail.google.com"),
+			res:   false,
+		},
+		{
+			e:     ParseEntry("*.google.com"),
+			other: ParseEntry("*.yahoo.com"),
+			res:   false,
+		},
+		{
+			e:     ParseEntry("*.google.com"),
+			other: ParseEntry("192.168.0.1"),
+			res:   false,
+		},
+		{
+			e:     ParseEntry("*.google.com"),
+			other: ParseEntry("192.168.0.1/24"),
+			res:   false,
+		},
+	}
+
+	for _, te := range tests {
+		assert.Equal(t, te.res, te.e.Equal(te.other), "test: %s equal %s", te.e, te.other)
+	}
+}
+
+func TestParseEntry(t *testing.T) {
+	var tests = []struct {
+		s   string
+		res Entry
+	}{
+		{
+			s:   "192.168.0.1",
+			res: &urlEntry{u: parseURL("192.168.0.1")},
+		},
+		{
+			s:   "192.168.0.1:80",
+			res: &urlEntry{u: parseURL("192.168.0.1:80")},
+		},
+		{
+			s:   "192.168.0",
+			res: &urlEntry{u: parseURL("192.168.0")},
+		},
+		{
+			s:   "192.168.0.1/24",
+			res: &cidrEntry{ipnet: &net.IPNet{IP: net.ParseIP("192.168.0.0"), Mask: net.CIDRMask(24, 32)}},
+		},
+		{
+			s:   "192.168.0/24",
+			res: &urlEntry{u: parseURL("192.168.0/24")},
+		},
+		{
+			s:   "*.google.com",
+			res: &urlEntry{u: parseURL("*.google.com")},
+		},
+		{
+			s:   "google.com:8080",
+			res: &urlEntry{u: parseURL("google.com:8080")},
```

Comment:
Not seeing any tests that exercise the `parseEntry` code that strips the Scheme, or entries that omit scheme being tested against those with one.